### ISSUE TITLE
InvalidCastException on startup. Need better error message.

### ISSUE THREAD

**Issue by StephaneBischoff** [Azure/azure-webjobs-sdk-script#313]
Hi,

Using latest alpha packages.  (WebJobs.Script 1.0.0-alpha1-10027)

Getting exception on startup. 
An unhandled exception of type 'System.InvalidCastException' occurred in Microsoft.Azure.WebJobs.Script.dll
Additional information: **Unable to cast object of type 'Newtonsoft.Json.Linq.JArray' to type 'Newtonsoft.Json.Linq.JObject'.**
   at Microsoft.Azure.WebJobs.Script.NodeFunctionDescriptorProvider.TryCreate(FunctionFolderInfo functionFolderInfo, FunctionDescriptor& functionDescriptor)
   at Microsoft.Azure.WebJobs.Script.ScriptHost.ReadFunctions(List`1 functionFolderInfos, IEnumerable`1 descriptorProviders)
   at Microsoft.Azure.WebJobs.Script.ScriptHost.ReadFunctions(ScriptHostConfiguration config, IEnumerable`1 descriptionProviders)
   at Microsoft.Azure.WebJobs.Script.ScriptHost.Initialize()
   at Microsoft.Azure.WebJobs.Script.ScriptHost.Create(ScriptHostConfiguration scriptConfig)
   at WebJobScriptRunner.Program.Main()

```
public class Program
    {
        static void Main()
        {
            ScriptHostConfiguration config = new ScriptHostConfiguration()
            {
                // path to root function script directory
                RootPath = Environment.CurrentDirectory
            };

            var host = ScriptHost.Create(config); <--- EXCEPTION THROWN HERE
            host.RunAndBlock();
        }
    }
```

I have no clue what I am doing wrong in my first Azure function project. Here is my host.jon and function.json.

`
{
  "id": "5e8ef5be0c6849819ef8b70ce0cb0d1e"
}
`

`
{
  "bindings": [
    {
      "direction": "in",
      "queueName": "script-jobs",
      "type": "queueTrigger"
    }
  ]
}
`

**Comment by mathewc** [comment]
Hi Stehane. This repo contains a functioning example showing how to start up and run a ScriptHost locally. See [here](https://github.com/Azure/azure-webjobs-sdk-script/blob/master/src/WebJobs.Script.Host/Program.cs). That example uses ScriptHostManager which provides additional capabilities (e.g. file watch for changes and auto-reload). Can you try using ScriptHostManager and see if you get the same result?

Also, can you point me to a repo that contains your code so I can help figure out what might be wrong?

**Comment by StephaneBischoff** [comment]
Thanks mathewc. I was able to compile the example provided. What I don't understand fully is where can I get the Microsoft.Azure.WebJobs.Script.ScriptHostManager class? Its not available in the Nuget package <package id="Microsoft.Azure.WebJobs.Script" version="1.0.0-alpha1-10027" targetFramework="net46" /> ?

**Comment by mathewc** [comment]
ScriptHostManager is publically exported by Microsoft.Azure.WebJobs.Script (see [here](https://github.com/Azure/azure-webjobs-sdk-script/blob/master/src/WebJobs.Script/Host/ScriptHostManager.cs)).

My plan is to go through the "run in console app" steps myself and document them in the wiki.

**Comment by StephaneBischoff** [comment]
Thanks mathewc. Is it possible that the Nuget package "Microsoft.Azure.WebJobs.Script" is not up to date? It does not contain the publically exported ScriptHostManager class.

**Comment by mathewc** [comment]
Yes it is. I'm updating it now, and verifying the whole end to end. I'll publish the steps in the wiki including a working sample when I'm finished, and I'll update this thread.

**Comment by mathewc** [comment]
Ok, after performing a bunch of package updates, I've got the **Microsoft.Azure.WebJobs.Script** package in shape. I was able to create a new Console app, reference this package, and get things running. I also deployed it as a continuous WebJob and that worked also. I've added a sample [here](https://github.com/Azure/azure-webjobs-sdk-script-samples/tree/master/WebJobScript) for this.

**Comment by StephaneBischoff** [comment]
Awesome. Thanks mathewc. I works now.

**Comment by mathewc** [comment]
Does the Dashboard work for you as well? I think there is another small fix that I need to make to get that working. Do you see your job invocations in the Dashboard?

**Comment by mathewc** [comment]
FYI - Dashboard issue is fixed in the latest package version 1.0.0-alpha1-10198.
